content/article/passing-by.md

```diff
@@ -0,0 +1,237 @@
+{
+   "image" : "/images/passing-by/relay.jpg",
+   "thumbnail" : "/images/passing-by/relay.jpg",
+   "date" : "2021-07-13T11:48:52",
+   "categories" : "development",
+   "title" : "Passing by",
+   "tags" : [
+      "reference",
+      "functions"
+   ],
+   "draft" : false,
+   "description" : "Looking at how arguments are passed to subs",
+   "authors" : [
+      "thibault-duponchelle"
+   ]
+}
+
+## Assignments by copy or reference
+When assigning, `perl` often copies the value, see the following code:
+```perl
+my $var1 = "not modified";
+
+my $var2 = $var1; # Copy the content, not the variable itself
+$var1 = "modified";
+
+print "$var2\n";
+```
+This code will print the initial *non-modified* value:
+```
+not modified
+```
+
+It is because the assignment copies the content of the variable and does not *alias* the variable. 
+
+To better understand, you can check by printing the [reference](https://perldoc.perl.org/perlref) of the variable:
+```perl
+# Compare storage location
+print \$var1 . " vs " . \$var2 . "\n";
+```
+That gives 2 different storage locations:
+```perl
+SCALAR(0x55589c6378e0) vs SCALAR(0x55589c6378f8)
+                   ^^                        ^^  
+```
+So it clearly states that these are 2 different variables.
```

Comment:
Two different scalar values (not variables).

Reply:
Are you sure? I'm refering to the *container* and In Mastering Perl (Symbol Tables - Typeglobs) you well define them as "variable"